extensions/azurePublishNew/src/types/index.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-regions
+export const LuisAuthoringSupportLocation = ['westus', 'australiaeast', 'westeurope'];
```

Comment:
Constants and enums should be separated into a constants.ts.  The types.ts file should contain only types as they will compile away.  If a component is the only usage of a type or constant then we should prefer to move it into that component's module.